mr/src/test/java/org/apache/iceberg/mr/hive/TestHiveIcebergStorageHandlerWithEngine.java

```diff
@@ -335,6 +335,25 @@ public void testInsertFromSelectWithOrderBy() throws IOException {
     HiveIcebergTestUtils.validateData(table, records, 0);
   }
 
+  @Test
+  public void testInsertFromSelectWithProjection() throws IOException {
+    Assume.assumeTrue("Tez write is not implemented yet", executionEngine.equals("mr"));
+
+    Table table = testTables.createTable(shell, "customers", HiveIcebergStorageHandlerTestUtils.CUSTOMER_SCHEMA,
+        fileFormat, ImmutableList.of());
+    testTables.createTable(shell, "orders", ORDER_SCHEMA, fileFormat, ORDER_RECORDS);
+
+    shell.executeStatement(
+        "INSERT INTO customers (customer_id, last_name) SELECT distinct(customer_id), 'test' FROM orders");
+
+    List<Record> expected = TestHelper.RecordsBuilder.newInstance(HiveIcebergStorageHandlerTestUtils.CUSTOMER_SCHEMA)
+        .add(0L, null, "test")
+        .add(1L, null, "test")
+        .build();
+
+    HiveIcebergTestUtils.validateData(table, new ArrayList<>(expected), 0);
```

Comment:
Why does this create a new `ArrayList` from the expected records?

Reply:
`HiveIcebergTestUtils.validateData` uses in place sort to compare the actual and the expected records. This is needed of other uses of the method, so I just recreated a new list for the test case

Reply:
Doesn't this leak the concern about sorting outside of the method? I would rather have the array copy happen defensively inside the method.

Reply:
Good point. Fixed here, and all the relevant other places